src/requests/pin_chat_message.rs

```diff
@@ -0,0 +1,45 @@
+use crate::requests::{ChatId, RequestContext, RequestFuture, ResponseResult, Request};
+use crate::network;
+
+/// Use this method to get up to date information about the chat 
+/// (current name of the user for one-on-one conversations, 
+/// current username of a user, group or channel, etc.). 
+/// Returns a Chat object on success.
+#[derive(Debug, Clone, Serialize)]
+pub struct PinChatMessage<'a> {
+    #[serde(skip_serializing)]
+    ctx: RequestContext<'a>,
+    /// Unique identifier for the target chat or username 
+    /// of the target supergroup or channel (in the format @channelusername)
+    chat_id: ChatId,
+    message_id: i32,
+    disable_notification: bool
+}
+
+impl<'a> PinChatMessage<'a> {
+    pub(crate) fn new(
+        ctx: RequestContext<'a>, chat_id: ChatId, message_id: i32
+    ) -> Self {
+        Self { ctx, chat_id, message_id, disable_notification: false }
+    }
+
+    pub fn disable_notification(mut self) -> Self {
+        self.disable_notification = true;
+        self
+    }
+}
+
+impl<'a> Request<'a> for PinChatMessage<'a> {
+    type ReturnValue = bool;
```

Comment:
Oh. All 'return True on success' methods should return `()` I think. (But it needs changes in parsing answer probably so we can leave a TODO for now)
```suggestion
    type ReturnValue = bool; // TODO: unit type
```

Reply:
I agree